logstash-core/src/main/java/org/logstash/instrument/witness/QueueWitness.java

```diff
@@ -0,0 +1,60 @@
+package org.logstash.instrument.witness;
+
+import org.logstash.instrument.metrics.gauge.TextGauge;
+
+/**
+ * Witness for the queue.
+ */
+final public class QueueWitness {
+
+    private final TextGauge type;
+    private final Snitch snitch;
+
+    /**
+     * Constructor.
+     */
+    public QueueWitness() {
+        type = new TextGauge("type");
+        snitch = new Snitch(this);
+    }
+
+    /**
+     * Get a reference to associated snitch to get discrete metric values.
+     *
+     * @return the associate {@link Snitch}
+     */
+    public Snitch snitch() {
+        return snitch;
+    }
+
+    /**
+     * Sets the type of the queue.
+     *
+     * @param type The type of the queue.
+     */
+    public void type(String type) {
```

Comment:
WDYT of event driven names for things? Rather than `type` which is basically a setter, something like `queueStart(String type)`. The rationale being that queue types only change when a queue is started (and really that can be hoisted up to when a pipeline is started). 

Given that the idea of the `Snitch` is that we don't want to have getters given that we don't have real setters (something I like), I think having setter-like things looks like a smell to me. 

It also makes me think we could use similar methods for pipelines (`pipelineStart`, `pipelineStop`). WDYT?